.github/workflows/publish_pure_python.yml

```diff
@@ -0,0 +1,80 @@
+name: Build and publish pure Python package
+
+on:
+  workflow_call:
+    inputs:
+      test_extras:
+        description: Any extras_requires modifier that should be used to install the package for testing
+        required: false
+        default: ''
+        type: string
+      test_command:
+        description: The command to run to test the package (will be run in a temporary directory)
+        required: false
+        default: ''
+        type: string
+      libraries:
+        description: Packages needed to build the source distribution for testing (installed using apt)
+        required: false
+        default: ''
+        type: string
+      upload_to_pypi:
+        description: Always upload to PyPI after successful builds - if false, only upload when pushing tags
+        required: false
+        default: false
+        type: boolean
+      repository_url:
+        description: The PyPI repository URL to use
+        required: false
+        default: ''
+        type: string
+      submodules:
+        description: Whether to checkout submodules
+        required: false
+        default: true
+        type: boolean
+    secrets:
+      pypi_password:
+        required: false
+
+jobs:
+
+  build:
+    name: Build source and wheel distribution
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          lfs: true
+          submodules: ${{ inputs.submodules }}
+      - name: Install dependencies
+        if: ${{ inputs.libraries != '' }}
+        uses: ConorMacBride/install-package@main
```

Comment:
I am happy to move this under OpenAstronomy as well if you want some help looking after it.